ERCS/erc-7599.md

```diff
@@ -0,0 +1,132 @@
+---
+eip: 7599
+title: On-chain Autonomous AI Agent Non-fungible Token
+description: A standard that makes NFT intelligent and interactive.
+author: Aiden X (@4ever9)
+discussions-to: https://ethereum-magicians.org/t/on-chain-autonomous-ai-agent-non-fungible-token/18194/1
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-01-18
+requires: 6551
+---
+
+## Abstract
+
+The proposal defines a set of standards for upgrading non-fungible tokens to on-chain AI Agent Account and specifying their interaction mechanisms. These AI Agent Accounts allow interaction with other smart contracts and off-chain applications.
+
+## Motivation
+
+Defining standards for on-chain AI Agent Account paves the path for a democratized human-AI collaborative society, allows for creations of AI data as a new class of asset, and treating each of the AI Agent Account as an autonomous yield-generating entity. Because it is compatible with ERC-6551 protocol, AI Agent Account is also an Ethereum Account. At the same time, many scenarios can be extended by defining `abilityURI`, interaction processes, and interfaces in AI Agent Account. Here are some examples of scenarios:
+
+- abilityURI links to a prompt: Users can chat with an AI Agent Account or ask an AI Agent Account to help draw a picture and mint it into an NFT.
+- abilityURI links to a langchain task: Users can ask an AI Agent Account to help with complex tasks.
+- abilityURI links to an ML algorithm: Let an AI Agent Account predict token prices or the weather
+
+An AI Agent Account must also define its interaction interface, and based on business requirements, this interface can be implemented for payment, whitelist, or qualification interactions, according to permission controls.
+
+## Overview
+
+The system extends the definition of Account in ERC-6551, which mainly includes:
+
+1. Add an `abilityURI` method, whose content may be an IPFS or s3 link, and the link content may be a prompt or a snippet of langchain code.
+2. Define an interface for how other contracts or external accounts interact with an AI Agent Account.
+3. Define an interface for receiving off-chain AI computation results.
+
+The following diagram shows how the protocol works:
+
+![../assets/erc-7599/arch.png](Arch)
+
+1. The off-chain AI Agent service gets the `abilityURL` through function `abilityURI()`, and then downloads the prompt or a snippet of langchain code through `abilityURL`.
+2. Other smart contracts or users request on-chain AI Agent jobs by invoking `requestAgent` function with user input as a parameter and the AI Agent Account emits the `AgentRequested` event.
+3. Off-chain AI Agent service listens to the `AgentRequested` event and invokes the AI model computation and then receives an execution response.
+4. Off-chain AI Agent service returns the result through `handleAgentResponse`.
+5. 
```

Comment:
```suggestion
```

This is not an allowed section. You could move it to a subheading under Specification if you like.